stl/inc/chrono

```diff
@@ -2209,6 +2221,2949 @@ namespace chrono {
+
+            if (_Leap != _LeapSecondRep::_None) {
+                if (_Hour_24 == 23 && _Minute == 59 && _Second >= 59) {
+                    // It's possible that the parsed time doesn't exist because (a) _Seconds == 60 and there *isn't* a
+                    // leap second insertion or (b) _Seconds == 59 and there *is* a leap second subtraction.
+
+                    // Check for quick exit.
+                    const auto& _Tzdb{_CHRONO get_tzdb()};
+                    if (_Leap == _LeapSecondRep::_Negative && _Tzdb._All_ls_positive) {
+                        return true;
+                    }
+
+                    const bool _Possible_insertion{_Second == 60};
+                    const sys_seconds _Target_sys_time{
+                        _CHRONO floor<seconds>(_Dur) + (_Possible_insertion ? seconds{0} : seconds{1})};
+                    const auto& _Ls_vector{_Tzdb.leap_seconds};
+                    const auto _It{_STD lower_bound(_Ls_vector.begin(), _Ls_vector.end(), _Target_sys_time)};
+                    const bool _Match_leap{_It != _Ls_vector.end() && *_It == _Target_sys_time};
+
+                    // Condition for a good parse: (_Seconds == 59 && !_Match_leap) || (_Match_leap &&
+                    // _It->_Is_positive()). Below is the inverse of this.
+                    if (!(_Match_leap ? _It->_Positive() : !_Possible_insertion)) {
+                        return false;
+                    }
+                }
+
+                if (_Leap == _LeapSecondRep::_All) {
+                    constexpr bool _Can_rep_sec = _Can_represent<_DurationType, seconds>();
+                    if constexpr (_Can_rep_sec) {
+                        _Dur = utc_clock::from_sys(sys_time<_DurationType>{_Dur}).time_since_epoch();
+                        // If we parsed a positive leap second, then _Dur, interpreted as a system time, refers to the
+                        // second *after* insertion. Need to back up one second in this case.
+                        if (_Second == 60) {
+                            _Dur -= _CHRONO duration_cast<_DurationType>(seconds{1});
+                        }
+
+                        if (_Original_leap == _LeapSecondRep::_File_time) {
+                            _Dur -= _CHRONO duration_cast<_DurationType>(
+                                filesystem::_File_time_clock::_Skipped_filetime_leap_seconds);
+                        }
+                    } else {
+                        // Error if _Dur can't represent the above adjustment.
+                        return false;
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        _NODISCARD bool _Make_day(day& _Day_result) {
+            if (_Used_fields() == _F_day) {
+                _Day_result = day{static_cast<unsigned int>(*_Day)};
+                return true;
+            } else {

```

Comment:
Nitpick: I think we can scratch the else and simply `return false;` here, which seems more in line with the rest of the code base. Feel free to ignore.

Appears below

Reply:
While we often use early return, I am inclined to leave this as-is - the rationale being that it wouldn't significantly save indentation, and it wouldn't save any lines (if we keep an empty line after the closing brace of the `if`, which we usually do). As written, the `return` statements are aligned, which seems reasonable.